### PR TITLE
feat: ✨Manifest Editor

### PR DESCRIPTION
**Pull request by KANAjetzt**
- Loading the *manifest.json* file of a new created mod
- Validating the fields while typing
- Saving the changes to the connected mods manifest.json file
- Saving the Default Configuration field via the Config Editor

### Known issues

> - Right now there is only a validation check while typing to a field. If this field is invalid the text will not be saved to the manifest. So if you save a invalid manifest you don't get an error it just saves the last valid version of it.

*Fixed with [d3aeefb](https://github.com/GodotModding/godot-mod-tool/pull/27/commits/d3aeefbff479de60d949b47c18595c1fe8d15d50)*
Added `is_valid` to the `StringInput` and added a check before saving if all Inputs are valid.
If not an error is logged to the output.

### Todos

- [x]  Implement validation check before saving 
*Implemented in [d3aeefb](https://github.com/GodotModding/godot-mod-tool/pull/27/commits/d3aeefbff479de60d949b47c18595c1fe8d15d50)*
- [x] Return an empty array instead of `[""]` if the input is empty
*fixed in [ffdbde8](https://github.com/GodotModding/godot-mod-tool/pull/27/commits/ffdbde890fa83c00f400e3327e6ebc514e00bcfd)*

### CODE REVIEW

File: addons/mod_tool/manifest_editor.gd [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+extends PanelContainer
+
+
+var input_fields := []
+
+onready var v_box := $ScrollContainer/VBox
+onready var input_name := $"%ModName"
+
+
+func _ready():
```
Comment (KANAjetzt) [inline addons/mod_tool/manifest_editor.gd:11]: `-> void:`

File: addons/mod_tool/manifest_editor.gd [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	_update_manifest_value(node, compatible_game_versions)
+
+
+func _on_Tags_input_text_changed(new_text: String, node: Node):
+	var tags := ModToolUtils.get_array_from_comma_separated_string(new_text)
+	_update_manifest_value(node, tags)
+
+
+func _on_SaveManifest_pressed():
+	save_manifest()
```
Comment (KANAjetzt) [inline addons/mod_tool/manifest_editor.gd:119]: `-> void:`

File: addons/mod_tool/manifest_editor.gd [hunk 3]
```diff
@@ -0,0 +1,119 @@
+tool
+extends PanelContainer
+
+
+var input_fields := []
+
+onready var v_box := $ScrollContainer/VBox
```
[the remark this answers is not in the document]
Comment (KANAjetzt) [reply addons/mod_tool/manifest_editor.gd]: fixed *[08a26d3](https://github.com/GodotModding/godot-mod-tool/pull/27/commits/08a26d3b7ba44ffab7f9449c104780ad4236a09b)*

File: addons/mod_tool/manifest_editor.gd (hunk 1 above)
Comment (KANAjetzt) [reply addons/mod_tool/manifest_editor.gd]: fixed - *[f6c0a81](https://github.com/GodotModding/godot-mod-tool/pull/27/commits/f6c0a811cee3f87db6d8f4888dba6d243a07887e)*

File: addons/mod_tool/manifest_editor.gd (hunk 2 above)
Comment (KANAjetzt) [reply addons/mod_tool/manifest_editor.gd]: fixed - *[f6c0a81](https://github.com/GodotModding/godot-mod-tool/pull/27/commits/f6c0a811cee3f87db6d8f4888dba6d243a07887e)*

File: addons/mod_tool/dock.gd [hunk 4]
```diff
@@ -14,18 +14,22 @@ onready var popup := $"%Popup"
...
+onready var manifest_editor = $"%ManifestEditor"
+onready var config_editor = $"%ConfigEditor"
 
 
 func _ready() -> void:
 	tab_parent_bottom_panel = get_parent().get_parent() as PanelContainer
 
-	_load_manifest()
-	_is_manifest_valid()
+	ModToolStore.label_output = label_output
```
[the remark this answers is not in the document]
Comment (KANAjetzt) [reply addons/mod_tool/dock.gd]: Deleted - *[7ef73a9](https://github.com/GodotModding/godot-mod-tool/pull/27/commits/7ef73a962d5dc2bc4799a3b783dae15faf9d2a42)*

**Review by Qubus0 — APPROVED** [review]
tested (3.5.2), works!
well done 👍

**Pull request merged**